Controllers/UserController.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Controllers;
+
+use Models\UserModel;
+use Models\ViewModel;
+
+class UserController
+{
+    private $userModel;
+    private $viewModel;
+
+    public function __construct()
+    {
+        $this->userModel = new UserModel();
+        $this->viewModel = new ViewModel();
+    }
+
+    public function loginAction()
+    {
+        if ($_SERVER['REQUEST_METHOD'] == 'POST' && isset($_POST)) {
+            $login = $_POST['login'];
+            $password = $_POST['password'];
+            $result = $this->userModel->login($login, $password);
+            switch ($result) {
+                case 1 :
+                    $this->viewModel->location('/');
+                    break;
+                case 2 :
+                    $errMsg = 'Мама пока не дала разрешение на вход!';
+                    $this->viewModel->render(null, $errMsg);
+                    break;
+                case 0 :
+                    $errMsg = 'Неверный логин или пароль!';
+                    $this->viewModel->render(null, $errMsg);
+            }
+        } else {
+            $errMsg = 'Простите, но что-то пошло не так. Попробуйте еще раз';
+            $this->viewModel->render(null, $errMsg);
+        }
+    }
+
+    public function logoutAction()
+    {
+        session_destroy();
+        $this->viewModel->location('/');
+    }
+
+    public function registrationAction($errMsg = null)
+    {
+        $isSigned = $this->userModel->isSigned();
+        if (!$isSigned) {
+            $options = [
+                'title' => 'Регистрация',
+                'content' => 'registration.phtml'
+            ];
+            $this->viewModel->render($options, $errMsg);
+        } else {
+            $this->viewModel->location('/');
+        }
+    }
+
+    public function newAction()
+    {
+        if ($_SERVER['REQUEST_METHOD'] == 'POST' && isset($_POST)) {
+            $result = $this->userModel->newUser($_POST);
+        }
+        if ($result === 'exist') {
+            $this->registrationAction('Пользователь с таким логином уже зарегистрирован.');
```

Comment:
better to determine what title do we have depends on our **$result** and finally call our registration action outside of if statement